plugins/media/wowza/services/LiveConversionProfileService.php

```diff
@@ -24,21 +24,39 @@ public function initService($serviceId, $serviceName, $actionName)
 
 	/**
 	 * Serve XML rendition of the Kaltura Live Transcoding Profile usable by the Wowza transcoding add-on
-	 * 
+	 *
 	 * @action serve
 	 * @param string $streamName the id of the live entry with it's stream suffix
 	 * @param string $hostname the media server host name
+	 * @param string $extraParams is a json object containing the stream parameters transfered by the encoder
 	 * @return file
-	 * 
+	 *
 	 * @throws KalturaErrors::ENTRY_ID_NOT_FOUND
 	 * @throws WowzaErrors::INVALID_STREAM_NAME
+	 * @throws KalturaErrors::INGEST_NOT_FOUND_IN_CONVERSION_PROFILE
 	 */
-	public function serveAction($streamName, $hostname = null)
+	public function serveAction($streamName, $hostname = null, $extraParams = null)
 	{
+		$streamParametersArray = array(
+			'streamName' => $streamName,
+			'hostname' => $hostname,
+			'audiodatarate' => 0,
+			'videodatarate' => 0,
+			'width' => 0,
+			'height' => 0,
+			'framerate' => 0,
+			'videocodecidstring' => "",
+			'audiocodecidstring' => ""
```

Comment:
videocodecidstring => videoCodecId why do you need the string postfix.
Same goes for audiocodecidstring

Also stay consistent with upper case in long variables you have "streamName" but also "videodatarate" applies for other long variables as well.